agent/hcp/client/client.go

```diff
@@ -120,6 +143,31 @@ func (c *hcpClient) PushServerStatus(ctx context.Context, s *ServerStatus) error
 	return err
 }
 
+func (c *hcpClient) InitMetricsClient(ctx context.Context, endpoint string) error {
+	hcpConfig, err := c.cfg.HCPConfig()
+	if err != nil {
+		return err
+	}
+
+	// TODO:In follow up PR (CC-4635) : Need to use oauth2 transport in the otlpmetrichttp client to add token.
+	// We likely need to fork and update the otlpmetrichttp client, as there is no interface to do this currently.
+	exp, err := otlpmetrichttp.New(ctx, otlpmetrichttp.WithEndpoint(endpoint), otlpmetrichttp.WithTLSClientConfig(hcpConfig.APITLSConfig()))
```

Comment:
Note for reviewers: I initially wanted to have this `otlpmetrichttp` client directly in the Exporter (telemetry/exporter.go), however, to do HCP auth, we need the `hcpConfig` which is injected during bootstrap into the HCP client.

With this design, it made mocking very easy, and so testing extremely easy as well. 

Other options I considered:
1. Return the `hcpConfig ` with a new `HCPConfig` method on the HCP Client and keep the OTLP specific client in Exporter. Although this works fine, I didn't get the mocking benefits, and it became hard to test the Exporter. A
2. Remove Exporter and use the Reporter + HCP Client only. This made the Reporter really complex. From a code maintainability standpoint, this felt much cleaner and the concerns are well separated.